libs/utils/include/utils/algorithm.h

```diff
@@ -59,7 +59,7 @@ template<typename T, typename = std::enable_if_t<std::is_unsigned<T>::value>>
 constexpr inline T ctz(T x) noexcept {
     static_assert(sizeof(T) * CHAR_BIT <= 64, "details::ctz() only support up to 64 bits");
     T c = sizeof(T) * CHAR_BIT;
-    x &= -x;    // equivalent to x & (~x + 1)
+    x &= (~x + 1);
```

Comment:
I would like to keep the old code here because I checked that it generated a better sequence on ARM. So Maybe use a #if for msvc.

Reply:
We were curious and tried to reproduce it on godbolt, but all compilers we tested generated the same code. But to keep all compilers happy I added the #if in 2944599e5974496b6da8964a1c058e80a6b1a41e

Reply:
okay good to know. thanks for looking into it.